src/helpers/chain.js

```diff
@@ -0,0 +1,379 @@
+  let key = roundData
+    ? "published.priceFeed." + feed + "_price_feed.latestRound"
+    : "published.priceFeed." + feed + "_price_feed";
+  return await vstorage.readLatest(key);
+};
+
+/**
+ * Function to get last 5 offers
+ * @param {Promise<import('@agoric/casting/src/follower-cosmjs').ValueFollower<T>>} follower offers and balances
+ * @returns {string} a list of offers
+ */
+export const getOffers = async (follower) => {
+  let history = [];
+  let counter = 0;
+  let lastVisited = -1;
+
+  for await (const followerElement of iterateReverse(follower)) {
+    if (counter === 5) {
+      break;
+    }
+
+    // If it is an offer status
+    if (followerElement.value.updated === "offerStatus") {
+      // Get id
+      let id = followerElement.value.status.id;
+
+      // If a new and final state
+      if (id !== lastVisited) {
+        // If it is not failed
+        if (!followerElement.value.status.hasOwnProperty("error")) {
+          history.push(followerElement.value);
+          counter++;
+        }
+        lastVisited = id;
+      }
+    }
+  }
+  return history;
+};
+
+/**
+ * Function to get the latest submitted round
+ * @param {string} oracle oracle address
+ * @returns {number} the latest round submitted
+ */
+export const getLatestSubmittedRound = async (oracle) => {
+  let fromBoard = makeFromBoard();
+  const unserializer = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+  const leader = makeLeader(networkConfig.rpcAddrs[0]);
+
+  const follower = await makeFollower(`:published.wallet.${oracle}`, leader, {
+    // @ts-expect-error xxx
+    unserializer,
+  });
+
+  // Get offers
+  let offers = await getOffers(follower);
+
+  return Number(
+    offers[0]["status"]["invitationSpec"]["invitationArgs"][0]["roundId"]
+  );
+};
+
+/**
+ * Function to check if submission was satisfied for a specific round
+ * @param {string} oracle oracle address
+ * @param {number} feedOfferId the offer id of the feed to check for
+ * @param {number} roundId the round Id which is checked
+ * @returns {boolean} whether a submission was successful for a specific round
+ */
+export const checkSubmissionForRound = async (oracle, feedOfferId, roundId) => {
+  let fromBoard = makeFromBoard();
+  const unserializer = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+  const leader = makeLeader(networkConfig.rpcAddrs[0]);
+
+  const follower = await makeFollower(`:published.wallet.${oracle}`, leader, {
+    // @ts-expect-error xxx
+    unserializer,
+  });
+
+  // Get offers
+  let offers = await getOffers(follower);
+
+  // Loop through offers starting from last offer
+  for (let i = 0; i < offers.length; i++) {
+    // Get current offer
+    let currentOffer = offers[i];
+
+    // If a price invitation and for the correct feed
+    let invitationType = currentOffer["status"]["invitationSpec"]["invitationMakerName"];
+    let previousOffer = currentOffer["status"]["invitationSpec"]["previousOffer"]
```

Comment:
```suggestion
    let invitationType = currentOffer["status"]["invitationSpec"][
    "invitationMakerName"];
    let previousOffer = currentOffer["status"]["invitationSpec"][
    "previousOffer"]
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/3c8d4afac6129f1b0a48c89a16cd53364ca93a4b